### PR TITLE
Added the new database tables for Job Avoidance

### PR DESCRIPTION
**Pull request by cjllanwarne**
Added the new database tables for Job Avoidance.

It may be better to do all Avoidance work in a branch rather than merging into DEVELOP one piece at a time.

### CODE REVIEW

**Comment by scottfrazer** [comment]
Can we add something in DataAccess like `getExecutionFromHash`?

File: src/main/migrations/changesets/job_avoidance.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            <column name="HASH" type="VARCHAR(32)">
+                <constraints nullable="true" />
+            </column>
+        </addColumn>
+
+        <addForeignKeyConstraint baseTableName="EXECUTION"
+                                 referencedTableName="EXECUTION"
+                                 baseColumnNames="AVOIDED_TO"
+                                 referencedColumnNames="EXECUTION_ID"
+
```
Comment (mcovarr) [inline src/main/migrations/changesets/job_avoidance.xml]: why the blank line?

File: src/main/migrations/changesets/job_avoidance.xml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.3.xsd">
+
+    <changeSet author="chrisl" id="job_avoidance">
+        <comment>Adding some tracking columns for determining eligibility for Job Avoidance.</comment>
+        <addColumn tableName="EXECUTION">
+            <column name="ALLOWS_AVOIDANCE" type="BOOLEAN" defaultValueBoolean="true" >
+                <constraints nullable="false" />
+            </column>
+            <column name="DOCKER_IMAGE_HASH" type="VARCHAR(32)">
```
Comment (mcovarr) [inline src/main/migrations/changesets/job_avoidance.xml]: is this large enough?

File: src/main/migrations/changesets/job_avoidance.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                <constraints nullable="true" />
+            </column>
+            <column name="AVOIDED_TO" type="INT">
+                <constraints nullable="true" />
+            </column>
+            <column name="EXECUTION_HASH" type="VARCHAR(32)">
+                <constraints nullable="true" />
+            </column>
+        </addColumn>
```
Comment (mcovarr) [inline src/main/migrations/changesets/job_avoidance.xml]: should we record the version of the algorithm used to generate the hash?

File: src/main/scala/cromwell/engine/db/slick/ExecutionComponent.scala [hunk 4]
```diff
@@ -15,6 +15,10 @@ case class Execution(workflowExecutionId: Int,
                      rc: Option[Int] = None,
                      startDt: Option[Timestamp] = None,
                      endDt: Option[Timestamp] = None,
+                     allowsAvoidance: Boolean = true,
+                     dockerImageHash: Option[String] = None,
+                     AvoidedTo: Option[Int] = None,
+                     ExecutionHash: Option[String] = None,
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/db/slick/ExecutionComponent.scala]: why are these capitalized?

File: src/main/migrations/changesets/job_avoidance.xml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        <createIndex indexName="HASH_INDEX"
+                     tableName="EXECUTION" unique="false">
+            <column name="EXECUTION_HASH"/>
+        </createIndex>
+
+        <addColumn tableName="SYMBOL">
+            <column name="HASH" type="VARCHAR(32)">
+                <constraints nullable="true" />
+            </column>
+        </addColumn>
```
Comment (mcovarr) [inline src/main/migrations/changesets/job_avoidance.xml]: same question here re hash algorithm version

File: src/main/migrations/changesets/job_avoidance.xml (hunk 2 above)
Comment (scottfrazer) [inline src/main/migrations/changesets/job_avoidance.xml]: I think it's just barely big enough for MD5 and will comfortably fit CRC, but no reason why it shouldn't be 100

**Comment by cjllanwarne** [comment]
Please don't comment any more here. Will re-open against Job Avoidance branch with changes made.

**Pull request closed without merging**